src/Functions/FunctionSQLJSON.h

```diff
@@ -260,12 +260,21 @@ class JSONValueImpl
         }
 
         if (status == VisitorStatus::Exhausted)
-        {
             return false;
-        }
 
         std::stringstream out; // STYLE_CHECK_ALLOW_STD_STRING_STREAM
-        out << current_element.getElement();
+        if (current_element.isString())
+        {
+            std::string_view element = current_element.getString();
+            if (element.starts_with('\"') && element.size() > 1)
+                out << element.substr(1, element.size() - 1);
```

Comment:
That's incorrect.

Try with 
```
milovidov-desktop :) SELECT JSONExtractString('{"test":"\uD83D\uDE0A"}')

SELECT JSONExtractString('{"test":"\\uD83D\\uDE0A"}')

Query id: a826b7a2-b134-44b8-81fa-7c58e83c7c43

┌─JSONExtractString('{"test":"\\uD83D\\uDE0A"}')─┐
│ {"test":"😊"}                                   │
└────────────────────────────────────────────────┘
```

You should get the smiley, not `\uD83D\uDE0A`

Reply:
Funny but it works without changing the code:
https://github.com/ClickHouse/ClickHouse/blob/824c5b27da9eccb239d216456599601108dea293/tests/queries/0_stateless/01889_sql_json_functions.reference#L28-L29

Reply:
Does it work correctly with
`{"data":"Hello \"World\" \\"}`
?

Reply:
Should return `Hello "World" \`

Reply:
And what will happen if you do

```
SELECT JSON_VALUE('{"hello":["\\uD83C\\uDF3A \\uD83C\\uDF38 \\uD83C\\uDF37 Hello, World \\uD83C\\uDF37 \\uD83C\\uDF38 \\uD83C\\uDF3A"]}', '$.hello');
```

It should return `["\uD83C\uDF3A \uD83C\uDF38 \uD83C\uDF37 Hello, World \uD83C\uDF37 \uD83C\uDF38 \uD83C\uDF3A"]`, not flowers.

Reply:
And for `{"data":["Hello \"World\" \\"]}`
it should return `["Hello \"World\" \\"]`.

Reply:
It doesn't but for a different reason.
And for JSONExtract `SELECT JSONExtractString('{"a":"Hello \"World\" \\"}', 'a') `  doesn't work too.
They both do not work with this case because they get document error, because in the string column with json string lies a string with lost slashes: `'{"test":"Hello "World" \"}'` => when trying to parse it, we get document error and return empty string.

Anyway I fixed to use ` readDoubleQuotedString` instead of `substr` when unquoting json_value result.

And both of them work if substitute that json with `'{"test":"Hello \\"World\\" \\\\"}'`.
So both `SELECT JSONExtractString('{"a":"Hello \\"World\\" \\\\"}', 'a')` and `JSON_VALUE('{"a":"Hello \\"World\\" \\\\"}', '$.a')` and will return ` Hello "World" \`.

And in mysql also `select JSON_VALUE('{"a": "Hello \"World\" \\"}', '$.a')` returns Null. But `select JSON_VALUE('{"a": "Hello \\"World\\" \\\\"}', '$.a')` returns `Hello "World" \\` -- one extra slash...

But in postgres `select jsonb_path_query('{"a": "Hello \"World\" \\"}', '$.a') #>> '{}'` returns `Hello "World" \` and with `'{"a": "Hello \\"World\\" \\\\"}'` has syntax error.

Such opposite results in mysql and postgres is confusing..

Reply:
Yes, I mean `SELECT JSONExtractString('{"a":"Hello \\"World\\" \\\\"}', 'a')`

`readDoubleQuotedString` is wrong, need `readJSONString`.
These methods use *slightly* different escaping.

Reply:
You are facing very subtle details about MySQL and PostgreSQL:

1. It depends on how do they parse string literals.
PostgreSQL does not recognize C-style backslash escapes. So no extra backslashes needed.
And PostgreSQL has additional syntax of `e'string'`

2. It depends on how do they present the strings in the user interface that you are using.
They may present the string already escaped for TSV or CSV while in fact it does not have extra escaping.

3. As these are subtle cases, we can expect that MySQL and PostgreSQL have bugs. ClickHouse should not have bugs.